actions/treasures.go

```diff
@@ -23,65 +21,32 @@ type treasureReq struct {
 }
 
 type treasureRes struct {
-	Success string `json:"success"`
+	Success bool `json:"success"`
 }
 
+var IotaWrapper = services.IotaWrapper
+var EthWrapper = services.EthWrapper
+
 // Verifies the treasure and claims such treasure.
 func (t *TreasuresResource) VerifyAndClaim(c buffalo.Context) error {
 	req := treasureReq{}
 	oyster_utils.ParseReqBody(c.Request(), &req)
 
 	addr := models.ComputeSectorDataMapAddress(req.GenesisHash, req.SectorIdx, req.NumChunks)
-	iotaAddr := make([]giota.Address, len(addr))
+	verify, err := IotaWrapper.VerifyTreasure(addr)
 
-	for i, address := range addr {
-		iotaAddr[i] = giota.Address(address)
+	if err != nil {
+		verify = false
 	}
 
-	transactionsMap, err := services.FindTransactions(iotaAddr)
-
-	if err != nil || len(transactionsMap) != len(iotaAddr) {
-		// indicate that PoW failure.
+	if verify == true {
+		//msg := EthWrapper.OysterCallMsg{From: "from", To: "to", Amount: 10, PrivateKey:, Gas: 1, TotalWei: 1, Data: [01, 11, 10]}
+		//result := EthWrapper.BuryPrl(msg)
```

Comment:
You'd be calling something like ClaimPRL here, not BuryPRL.  But I'm going to go ahead and approve.

Reply:
@pzhao5 @tenisakb ^

Reply:
I know we are doing claim but in the past messages was bury.. but understand, will repair